cmd/orca-tasks/task_watcher.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2019, Google, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package orca_tasks
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spinnaker/spin/cmd/gateclient"
+)
+
+// WaitForSuccessfulTask observes an Orca task to see if it completed successfully.
+func WaitForSuccessfulTask(gateClient *gateclient.GatewayClient, taskRef map[string]interface{}, maxAttempts int) error {
+	id := idFromTaskRef(taskRef)
+	task, resp, err := gateClient.TaskControllerApi.GetTaskUsingGET1(gateClient.Context, id)
+
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return fmt.Errorf("Encountered an error saving application, status code: %d\n", resp.StatusCode)
+	}
+	if !taskSucceeded(task) {
+		return fmt.Errorf("Encountered an error saving application, task output was: %v\n", task)
+	}
+
+	attempts := 0
+	for (task == nil || !taskCompleted(task)) && attempts < maxAttempts {
+		id := idFromTaskRef(taskRef)
+		task, resp, err = gateClient.TaskControllerApi.GetTaskUsingGET1(gateClient.Context, id)
```

Comment:
Is it deliberate that we don't check `err` or `respStatusCode` here?  (Maybe to just try again if there's a communication error?)

Reply:
`task == nil` handles the case where there is no retrieved task for any reason (network issues etc) which is why we're ignoring `resp` and `err` here.

Reply:
Makes sense :+1: